### DISCUSSION TITLE
Saving in a learnr tutorial

### DISCUSSION THREAD

**Discussion by EricFriedlander** [rstudio/learnr#638] [Q&A]
I am planning to teaching a statistics course this spring, making heavy use of learnr tutorials during class time. I'm thinking about leaving empty boxes in each tutorial so that students can use them as their notes. If I did this, it would be important that their progress (and thus notes) not get lost. I plan on putting these tutorials in a package hosted on Github and will have students update the package at the beginning of each week. How reliable is learnr with saving progress? What would cause progress to be lost?

**Comment by gadenbuie** [comment]
Hi @EricFriedlander ... how are you thinking of including the empty boxes? learnr only stores the state of its own responses, but not regular Shiny inputs, so you'd probably need to use an element like `question_text()` for those boxes. And students would need to "submit" their notes once for it to be stored and persisted between sessions.

I don't think that would be a very good fit for what you want, though. You might want to check out a service like https://web.hypothes.is/ which does something similar.
